### PR TITLE
user and form api's implemented

### PR DESCRIPTION
**Pull request by Dimundra**

### CODE REVIEW

File: server/routes/routes.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const userRoutes = require('./userRoutes');
+const formRoutes = require('./formRuotes');
 const testRoutes = require('./testRoutes');
```
Comment (kibernetika) [inline server/routes/routes.js:3]: You could remove Test rout and handler

File: server/services/auth.js [hunk 2]
```diff
@@ -1,22 +1,73 @@
...
-        expiresIn: expirationTimeOfJWT,
-      });
-      return { token: token };
-    }
+/// use user model
+const { user: userModel } = db.sequelize.models;
+///
+
+async function authenticateLogin(email, password) {
+  let user = await userModel.findOne({
```
Comment (kibernetika) [inline server/services/auth.js:11]: You can use method findByPk(121212)

File: server/services/auth.js [hunk 3]
```diff
@@ -1,22 +1,73 @@
...
+  const userWithSuchEmail = await userModel.findOne({
+    where: {
+      email: email,
+    },
+  });
+  if (userWithSuchEmail) {
+    return 'User with such email already exists!';
+  }
+
+  const userWithSuchPassword = await userModel.findOne({
```
Comment (kibernetika) [inline server/services/auth.js:50]: Remove this

File: server/services/auth.js [hunk 4]
```diff
@@ -1,22 +1,73 @@
...
+    nickname,
+    email,
+    password,
+  });
+
+  const payloadForJWT = { id: newUserId };
+  const expirationTimeOfJWT = '30000ms'; // 30 sec
+  const token = JWT.sign(payloadForJWT, secret, {
+    expiresIn: expirationTimeOfJWT,
+  });
```
Comment (kibernetika) [inline server/services/auth.js:69]: Ned move to separate service (method). Like a buildToken

**Review by kibernetika — CHANGES_REQUESTED** [review]

File: server/controllers/formController.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+// use form model
+const { form: formModel } = db.sequelize.models;
+///
+
+const getAllFormsHandler = async (request, h) => {
+  let forms = await formModel.findAll();
+
+  if (forms.length === 0) {
+    return Boom.notFound('No forms found!');
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:12]: Done!

File: server/controllers/formController.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  } else {
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:13]: Done!

File: server/controllers/formController.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+const getFormByIdHandler = async (request, h) => {
+  let form = await formModel.findOne({
+    where: {
+      id: request.params.id,
+    },
+  });
+
+  if (!form) {
+    return Boom.notFound("User with such id wasn't found!");
+  } else {
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:28]: Done!

File: server/controllers/formController.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+  await formModel.create({ name, data, userId });
+
+  return 'Form successfully added!';
+};
+
+const updateFormHandler = async (request, h) => {
+  const { name, data, userId } = request.payload;
+
+  const form = await formModel.findOne({
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:54]: Done!

File: server/controllers/formController.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    where: {
+      id: request.params.id,
+    },
+  });
+
+  if (!form) {
+    return Boom.notFound("Form with such id wasn't found!");
+  } else {
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:62]: Done

File: server/controllers/formController.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+          id: request.params.id,
+        },
+      }
+    );
+    return 'Form successfully updated!';
+  }
+};
+
+const deleteFormHandler = async (request, h) => {
+  const form = await formModel.findOne({
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/controllers/formController.js:76]: I will skip this query! Will do delete query directly!

File: server/services/tokenValidation.js [hunk 11]
```diff
@@ -1,15 +1,17 @@
...
+    where: {
+      id: decoded.id,
+    },
+  });
 
-  return { isValid: true };
+  if (!user) {
+    return { isValid: false };
+  } else {
+    return { isValid: true };
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/services/tokenValidation.js:13]: Done!

File: server/routes/userRoutes.js [hunk 12]
```diff
@@ -2,18 +2,46 @@ const Joi = require('joi');
...
+  options: {
+    validate: {
+      payload: Joi.object({
+        nickname: Joi.string().required('required!'),
+        email: Joi.string().required('required!').email(),
+        password: Joi.string()
+          .required('required')
+          .pattern(new RegExp(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/))
+          .min(6)
+          .max(20),
```
[the remark this answers is not in the document]
Comment (Dimundra) [reply server/routes/userRoutes.js:44]: Done!

File: server/routes/routes.js (hunk 1 above)
Comment (Dimundra) [reply server/routes/routes.js:3]: Done!

**Pull request merged**